contracts/oracles/default/UmbrellaPriceOracle.sol

```diff
@@ -0,0 +1,116 @@
+    UMBRELLA_FEEDS_ADDRESS = IUmbrellaFeeds(umbrellaFeeds);
+  }
+
+  /**
+   * @dev Admin-only function to set price feeds.
+   * @param underlyings Underlying token addresses for which to set price feeds.
+   * @param feeds The Oracle price feed contract addresses for each of `underlyings`.
+   */
+  function setPriceFeeds(address[] memory underlyings, string[] memory feeds) external onlyOwner {
+    // Input validation
+    require(
+      underlyings.length > 0 && underlyings.length == feeds.length,
+      "Lengths of both arrays must be equal and greater than 0."
+    );
+
+    // For each token/feed
+    for (uint256 i = 0; i < underlyings.length; i++) {
+      address underlying = underlyings[i];
+      // Set feed and base currency
+      priceFeeds[underlying] = feeds[i];
+    }
+  }
+
+  /**
+   * @dev Internal function returning the price in ETH of `underlying`.
+   * Assumes price feeds are 8 decimals!
+   * https://docs.fluxprotocol.org/docs/live-data-feeds/fpo-live-networks-and-pairs#mainnet-2
+   */
+  function _price(address underlying) internal view returns (uint256) {
+    // Get token/ETH price from feed
+    string memory feed = priceFeeds[underlying];
+    require(bytes(feed).length != 0, "No Umbrella price feed found for this underlying ERC20 token.");
+
+    // Get the NATIVE/USD price feed from NativeUSDPriceOracle
+    // 8 decimals are used
+    IUmbrellaFeeds.PriceData memory nativeTokenUsdPriceData = UMBRELLA_FEEDS_ADDRESS.priceData(NATIVE_TOKEN_USD_KEY);
+    uint256 nativeTokenUsdPrice = uint256(nativeTokenUsdPriceData.price);
+
+    if (nativeTokenUsdPriceData.price <= 0) return 0;
+    // 8 decimals are used
+    IUmbrellaFeeds.PriceData memory priceData = UMBRELLA_FEEDS_ADDRESS.priceData(feed);
+    uint256 tokenPrice = uint256(priceData.price);
+    // Flux price feed is 8 decimals:
+    return tokenPrice >= 0 ? (tokenPrice * 1e28) / uint256(nativeTokenUsdPrice) : 0;
+  }
+
+  /**
+   * @dev Returns the price in ETH of `underlying` (implements `BasePriceOracle`).
+   */
+  function price(address underlying) external view override returns (uint256) {
+    return _price(underlying);
+  }
+
+  /**
+   * @notice Returns the price in ETH of the token underlying `cToken`.
+   * @dev Implements the `PriceOracle` interface for Fuse pools (and Compound v2).
+   * @return Price in ETH of the token underlying `cToken`, scaled by `10 ** (36 - underlyingDecimals)`.
+   */
+  function getUnderlyingPrice(ICErc20 cToken) external view override returns (uint256) {
+    // Get underlying token address
+    address underlying = cToken.underlying();
+
+    // Get price
+    uint256 oraclePrice = _price(underlying);
+
+    // Format and return price
+    uint256 underlyingDecimals = uint256(ERC20Upgradeable(underlying).decimals());
+    return
+      underlyingDecimals <= 18
+        ? uint256(oraclePrice) * (10**(18 - underlyingDecimals))
+        : uint256(oraclePrice) / (10**(underlyingDecimals - 18));
```

Comment:
here `oraclePrice` is assumed to be scaled by `1e18` but the `_price()` call returns a price scaled by `1e28` .. is that intended?

Reply:
Good point. The 28 decimals scaling was intended to be used when calling `price` on the nativeUsd against the MPO.... So that the 28 + 8 / 18 = 18 decimals. This is indeed wrong.